airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/client/api.py

```diff
@@ -66,25 +72,79 @@ def state(self):
     def state(self, value):
         self._state = pendulum.parse(value[self.state_pk])
 
-    def __init__(self, *args, **kwargs):
+    def __init__(self, *args, include_deleted=False, **kwargs):
         super().__init__(*args, **kwargs)
+        self._include_deleted = include_deleted
         self._state = None
 
+    def _build_params(self, params: Mapping[str, Any] = None) -> Iterator:
+        """Build complete params for request"""
+        result_params = super()._build_params(params)
+        for filters in self._build_filters():
+            result_params.setdefault("filtering", []).extend(filters)
+
+            yield result_params
+
+    def _build_filters(self) -> Iterator[List[Mapping[str, Any]]]:
+        """Build complete filters"""
+        for delete_filter in self._delete_filters():
```

Comment:
if we don't have delete filters, shouldn't we still be getting state filters?

Reply:
we always have values to return, it is either an empty filter `{}` or list of up to 3 filters `{..}, {..}, {...}`